File: ProductNumDigits_RecusriveApproach/Product.py
"""
    Author : Pakshal Shashikant Jain 
    Date : 9/5/2024
    Program : Write a recursive program which accept number from user and return its
              product of digits.

              Input : 523
              Output : 30 
"""

########################################################################################################################################################

class Digits :
    #Characteristics of Class
    ino = 0;
    imult = 0;

    #Constructor of Class
    def __init__(self,value) :
        self.ino = value;
        self.imult = 1;

    #Behaviour of Class
    def Mult(self) :
        Digits = 0;

        Digits = self.ino % 10;
        self.imult = self.imult * Digits;
        self.ino = self.ino // 10;

        if(self.ino != 0) :
            self.Mult();

        return self.imult;

#########################################################################################################################################################

#Main Function
def main() :
    ino = 0;
    iret = 0;
    
    print("Jay Ganesh....");

    print("Enter One Number : ");
    ino = int(input());

    #Creating object of Digits Class
    dobj = Digits(ino);
    iret = dobj.Mult();

    #Print Products Calculated
    print("Products of Digits of Enter Number is : %d"%iret);

##########################################################################################################################################################################

#Call Back Method To Call Main Function
if __name__ == "__main__" :
    #Call To Main Function
    main();

File: ProductNumDigits_RecusriveApproach/Product.go
/*
    Author : Pakshal Shashikant Jain 
    Date : 9/5/2024
    Program : Write a recursive program which accept number from user and return its
              product of digits.

              Input : 523
              Output : 30 
*/

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
package main;
import "fmt";

var imult int = 1;

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To Calculate Products of Digits
func Mult(ino int) int {
	var Digits int = 0;

	Digits = ino % 10;
	imult = imult * Digits;
	ino = ino / 10;

	if(ino != 0) {
		Mult(ino);
	}

	return imult;
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
func main() {
	var ino int = 0;
	var iret int = 0;

	fmt.Println("Jay Ganesh......");
	
	//Taking Input From User
	fmt.Println("Enter One Number : ");
	fmt.Scan(&ino);

	//Call To Mult Function
	iret = Mult(ino);

	//Printing Products Calculated
	fmt.Println("Products of Digits is :",iret);
}